stdlib/src/memory/arc.mojo

```diff
@@ -101,6 +101,15 @@ struct Arc[T: Movable](CollectionElement):
             value^
         )
 
+    fn __init__(inout self, *, other: Self):
+        """Copy the object.
```

Comment:
**Suggestion** For `Arc`, what "the object" refers to could be somewhat ambiguous.

We should expand on this description to clarify that this is a cheap reference count increment copy, not a "deep" copy of the inner object.